### PR TITLE
feat(auth): persist active connection and add more docs

### PR DESCRIPTION
**Pull request by JadenSimon**
## Problem
The Auth service doesn't remember the active connection across sessions

## Solution
Persist the connection id and restore it later.

Originally I was going to implement something similar to `canAutoConnect` but just persisting the current connection is simpler and more user-friendly for the case of expired connections

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/credentials/auth.ts [hunk 1]
```diff
@@ -276,7 +356,24 @@ export class Auth implements AuthService, ConnectionManager {
         return connections.find(c => c.id === connection.id)
     }
 
-    private async updateState(id: Connection['id'], connectionState: ProfileMetadata['connectionState']) {
+    private async invalidateConnection(id: Connection['id']) {
```
Comment (justinmk3) [inline src/credentials/auth.ts:359]: A docstring explaining what this does would be helpful, because the details matter and can be tricky. E.g. does it make a backend request to invalidate, which things does it clear locally, etc.

File: src/credentials/sso/clients.ts [hunk 2]
```diff
@@ -161,6 +162,11 @@ export class SsoClient {
         }
     }
 
+    public async logout(request: Omit<LogoutRequest, OmittedProps> = {}) {
+        const method = this.client.logout.bind(this.client)
+        await this.call(method as ExtractOverload<typeof method>, request)
```
Comment (justinmk3) [inline src/credentials/sso/clients.ts:167]: this bind() + ExtractOverload pattern is hard to follow

**Review by justinmk3 — APPROVED** [review]

File: src/credentials/sso/clients.ts (hunk 2 above)
Comment (JadenSimon) [reply src/credentials/sso/clients.ts:167]: It's kind of a TS hack specific to JS SDK v3. Can probably make it nicer.

**Comment by JadenSimon** [comment]
I'll address the feedback in the next PR
